pkg/cloudprovider/cloudprovider.go

```diff
@@ -109,22 +109,22 @@ func (c *CloudProvider) Create(ctx context.Context, nodeClaim *karpv1.NodeClaim)
 		return nil, cloudprovider.NewNodeClassNotReadyError(stderrors.New(nodeClassReady.Message))
 	}
 	if nodeClassReady.IsUnknown() {
-		return nil, fmt.Errorf("resolving NodeClass readiness, NodeClass is in Ready=Unknown, %s", nodeClassReady.Message)
+		return nil, cloudprovider.NewNodeClassNotReadyError(fmt.Errorf("resolving NodeClass readiness, NodeClass is in Ready=Unknown, %s", nodeClassReady.Message))
 	}
 	if _, err = nodeClass.GetKubernetesVersion(); err != nil {
 		return nil, err
 	}
 
 	instanceTypes, err := c.resolveInstanceTypes(ctx, nodeClaim, nodeClass)
 	if err != nil {
-		return nil, fmt.Errorf("resolving instance types, %w", err)
+		return nil, cloudprovider.NewInsufficientCapacityError(fmt.Errorf("resolving instance types, %w", err))
 	}
 	if len(instanceTypes) == 0 {
 		return nil, cloudprovider.NewInsufficientCapacityError(fmt.Errorf("all requested instance types were unavailable during launch"))
 	}
 	instancePromise, err := c.instanceProvider.BeginCreate(ctx, nodeClass, nodeClaim, instanceTypes)
 	if err != nil {
-		return nil, fmt.Errorf("creating instance, %w", err)
+		return nil, cloudprovider.NewInsufficientCapacityError(fmt.Errorf("creating instance, %w", err))
```

Comment:
This won't necessarily be insufficient capacity, should be a more generic error

Reply:
Changed this to be the NewCreateError.
Added Reason Constant to be more specific.